aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -296,6 +330,7 @@ public Builder variable(@NonNull String key, String type, Object value) {
         public <R> AppSyncGraphQLRequest<R> build() throws AmplifyException {
             Objects.requireNonNull(this.operation);
             Objects.requireNonNull(this.responseType);
+            Objects.requireNonNull(this.modelSchemaRegistry);
```

Comment:
I see that `AppSyncRequestFactory` (used by `DataStore`) has been updated to pass this new required parameter.  However, it looks like `AppSyncGraphQLRequestFactory` (used by `API`) is not updated yet.  I think that needs to be updated as well, otherwise, this is going to throw an NPE for any usage of the API category.

Reply:
Hmm, actually I guess it won't throw an NPE since I see you are initing it to an empty registry in the `Builder` constructor.   So I guess that's fine, but we may need to update `AppSyncGraphQLRequestFactory` in the future, if Flutter is going to depend on it for their API category?